internal/resources/otelcollector/core.go

```diff
@@ -21,9 +20,9 @@ import (
 )
 
 // applyCommonResources applies resources to gateway and agent deployment node
-func applyCommonResources(ctx context.Context, c client.Client, name types.NamespacedName, rbac rbac, allowedPorts []int32) error {
+func applyCommonResources(ctx context.Context, c client.Client, name types.NamespacedName, component string, rbac rbac, allowedPorts []int32) error {
```

Comment:
please either change the name of the newly introduced `component` argument, or add some documentation what it is used for. All other arguments are self explanatory but component is not commonly known